src/libs/ReportUtils.ts

```diff
@@ -1437,13 +1438,13 @@ function getDisplayNameForParticipant(accountID?: number, shouldUseShortForm = f
         // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
         return formattedLogin;
     }
-    const longName = personalDetails.displayName ? personalDetails.displayName : formattedLogin;
-
-    const shortName = personalDetails.firstName ? personalDetails.firstName : longName;
 
+    const longName = PersonalDetailsUtils.getDisplayNameOrDefault(personalDetails, formattedLogin, false);
     if (!longName && shouldFallbackToHidden) {
```

Comment:
I'm confused why this login would still be needed since `PersonalDetailsUtils.getDisplayNameOrDefault` has a fallback to `common.hidden` built-in...

Seems like just doing this would be equivalent:

```js
const longName = PersonalDetailsUtils.getDisplayNameOrDefault(personalDetails, formattedLogin, shouldFallbackToHidden);
```

Reply:
I wanted to do the same, but the logic in both cases differs a bit.

If we use the suggested change:

```js
const longName = PersonalDetailsUtils.getDisplayNameOrDefault(personalDetails, formattedLogin, shouldFallbackToHidden);
const shortName = personalDetails.firstName ? personalDetails.firstName : longName;

return shouldUseShortForm ? shortName : longName;
```

We will calculate the `shortName` regardless of `shouldFallbackToHidden`, and finally will make a decision based on `shouldUseShortForm`: 

```js
return shouldUseShortForm ? shortName : longName;
```

However, if we use the approach I implemented (it's the same as it was in main):

We first build the `longName`, and then check: if longName is empty **and** `shouldFallbackToHidden`, we early return "Hidden". And only if `shouldFallbackToHidden` is `false`, we proceed to building the `shortName`.

So in this second case, the `shouldFallbackToHidden` takes priority over `shouldUseShortForm`.
This is how it was implemented in `main` and I didn't want to risk changing this approach. But in general, I would also prefer to use your suggested change. It seems highly unlikely to me that the long name will be empty while the short name will be non-empty. And even if so – it's better to use `shortName` instead of `Hidden` if that's the last available option.

Based on this, would you still approve that we make your suggested change? @roryabraham

Reply:
Oh, I see. What about this?

```ts
const longName = PersonalDetailsUtils.getDisplayNameOrDefault(personalDetails, formattedLogin, shouldFallbackToHidden);

// If the user name should be hidden, make sure we return `hidden` instead of the short name
if (shouldFallbackToHidden && longName === Localize.translateLocal('common.hidden')) {
    return longName;
}

const shortName = personalDetails.firstName ? personalDetails.firstName : longName;
return shouldUseShortForm ? shortName : longName;
```

Reply:
I think that makes the logic a bit clearer so that future devs are less likely to make the same mistake I made and assume that we could just seamlessly swap in `PersonalDetailsUtils.getDisplayNameOrDefault`

Reply:
Yes, this does look a bit clearer, thanks. I'll commit now

Reply:
Thanks, done ✔️